File: 151002/Saprin/RV/src/main/java/by/bsuir/rv/service/issue/IIssueService.java
package by.bsuir.rv.service.issue;

import by.bsuir.rv.dto.IssueRequestTo;
import by.bsuir.rv.dto.IssueResponseTo;
import by.bsuir.rv.exception.EntititesNotFoundException;
import by.bsuir.rv.exception.EntityNotFoundException;

import java.math.BigInteger;
import java.util.List;

public interface IIssueService {
    List<IssueResponseTo> getIssues() throws EntititesNotFoundException;
    IssueResponseTo getIssueById(BigInteger id) throws EntititesNotFoundException, EntityNotFoundException;
    IssueResponseTo addIssue(IssueRequestTo issue) throws EntititesNotFoundException, EntityNotFoundException;
    void deleteIssue(BigInteger id) throws EntityNotFoundException;
    IssueResponseTo updateIssue(IssueRequestTo issue) throws EntityNotFoundException, EntititesNotFoundException;

}

File: 151004/Bordilovsky/DC_Kafka&Redis/Publisher/Services/Interfaces/IIssueService.cs
using DC_REST.DTOs.Request;
using DC_REST.DTOs.Response;

namespace DC_REST.Services.Interfaces
{
    public interface IIssueService
    {
        IssueResponseTo CreateIssue(IssueRequestTo issueRequestDto);
        IssueResponseTo GetIssueById(int id);
        List<IssueResponseTo> GetAllIssues();
        IssueResponseTo UpdateIssue(int id, IssueRequestTo issueRequestDto);
        bool DeleteIssue(int id);
    }
}
